### PR TITLE
Make date time formatting parametric

### LINKED ISSUE #1349: Use user settings for formatting times and dates
We may want to cache this or something for performance when rendering time entries, etc.?

The setting should be respected everywhere in the app. Maybe a list should be made.

### PR DESCRIPTION
**Pull request by simonrozsival**
Closes #1349 

The date time converters now accept a parameter which states some of the supported formats. User's preferences are taken and passed to the converters in the defined places (see #1380)

Squash whenever

### CODE REVIEW

**Comment by simonrozsival** [comment]
I will update this to use #1539 once it is merged.

**Comment by simonrozsival** [comment]
Now this PR is waiting for #1550

**Comment by simonrozsival** [comment]
This is now ready for review.

File: Toggl.Foundation.MvvmCross/Converters/DateToTitleStringValueConverter.cs [hunk 1]
```diff
@@ -8,6 +8,15 @@ namespace Toggl.Foundation.MvvmCross.Converters
     [Preserve(AllMembers = true)]
     public sealed class DateToTitleStringValueConverter : MvxValueConverter<DateTimeOffset, string>
     {
+        private MvxValueConverter<DateTimeOffset, string> fallbackConverter;
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Foundation.MvvmCross/Converters/DateToTitleStringValueConverter.cs]: Also: I think `dateToStringConverter` would be a better name.

File: Toggl.Foundation.MvvmCross/Converters/DateTimeToTimeValueConverter.cs [hunk 2]
```diff
@@ -8,7 +9,22 @@ namespace Toggl.Foundation.MvvmCross.Converters
     [Preserve(AllMembers = true)]
     public sealed class DateTimeToTimeValueConverter : MvxValueConverter<DateTimeOffset, string>
     {
+        private TimeZoneInfo timeZone;
+
+        public DateTimeToTimeValueConverter(TimeZoneInfo timeZone)
+        {
+            this.timeZone = timeZone;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Converters/DateTimeToTimeValueConverter.cs]: Ensure not null

File: Toggl.Foundation.MvvmCross/Converters/DateTimeToTimeValueConverter.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Globalization;
+using System.Collections.Generic;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Converters/DateTimeToTimeValueConverter.cs]: 🔥

File: Toggl.Foundation.Tests/MvvmCross/Converters/DateTimeToTimeValueConverterTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+using System.Linq;
+using FluentAssertions;
+using FsCheck;
+using MvvmCross.Platform.Converters;
+using Toggl.Foundation.MvvmCross.Converters;
+using Xunit;
+
+namespace Toggl.Foundation.Tests.MvvmCross.Converters
+{
+    public class DateTimeToTimeValueConverterTests
```
Comment (arturs105) [inline Toggl.Foundation.Tests/MvvmCross/Converters/DateTimeToTimeValueConverterTests.cs]: Seal

File: Toggl.Foundation.Tests/MvvmCross/Converters/DateTimeToTimeValueConverterTests.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        private readonly DateTimeToTimeValueConverter converter = new DateTimeToTimeValueConverter(TimeZoneInfo.Local);
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(1)]
+        [InlineData(1L)]
+        [InlineData('a')]
+        public void FailsIfTheParameterIsNotAString(object parameter)
```
Comment (arturs105) [inline Toggl.Foundation.Tests/MvvmCross/Converters/DateTimeToTimeValueConverterTests.cs]: Enclose these tests in `class TheConvertMethod`

**Comment by simonrozsival** [comment]
This PR will be rewritten after #1597 is finished.

**Pull request closed without merging**